File: snippets/rust/src/connecting_lsp.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_core::*;

async fn get_lsp_info(sdk: Arc<BreezServices>) -> Result<LspInformation> {
    // ANCHOR: get-lsp-info
    let lsp_id = sdk.lsp_id().await?;
    let lsp_info = sdk.lsp_info().await?;
    // ANCHOR_END: get-lsp-info

    Ok(lsp_info)
}

async fn list_lsps(sdk: Arc<BreezServices>) -> Result<()> {
    // ANCHOR: list-lsps
    let available_lsps = sdk.list_lsps().await?;
    // ANCHOR_END: list-lsps

    Ok(())
}

async fn connect_lsp(sdk: Arc<BreezServices>, lsp_id: String) -> Result<()> {
    // ANCHOR: connect-lsp
    sdk.connect_lsp(lsp_id).await?;
    // ANCHOR_END: connect-lsp

    Ok(())
}

File: snippets/go/connecting_lsp.go
package example

import (
	"log"
)

func GetLspInfo() error {
	// ANCHOR: get-lsp-info
	lspId, err := sdk.LspId()
	if err != nil {
		return err
	}
	if lspId != nil {
		log.Printf("%#v", *lspId)
	}

	lspInfo, err := sdk.LspInfo()
	if err != nil {
		return err
	}
	log.Printf("%#v", lspInfo)
	// ANCHOR_END: get-lsp-info
	return nil
}

func ListLsps() error {
	// ANCHOR: list-lsps
	lspList, err := sdk.ListLsps()
	if err != nil {
		return err
	}
	log.Printf("%#v", lspList)
	// ANCHOR_END: list-lsps
	return nil
}

func ConnectLsp() error {
	// ANCHOR: connect-lsp
	lspId := "your selected lsp id"

	err := sdk.ConnectLsp(lspId)
	if err != nil {
		return err
	}
	// ANCHOR_END: connect-lsp
	return nil
}
